src/Text.Analyzers/Core/CodeAnalysisDictionary.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace Text.Analyzers
+{
+    /// <summary>
+    /// Source for "recognized" misspellings and "unrecognized" spellings obtained by parsing either
+    /// XML or DIC code analysis dictionaries.
+    /// </summary>
+    /// <Remarks>
+    /// <seealso href="https://docs.microsoft.com/en-us/visualstudio/code-quality/how-to-customize-the-code-analysis-dictionary?view=vs-2019"/>
+    /// </Remarks>
+    internal class CodeAnalysisDictionary
+    {
+        protected CodeAnalysisDictionary(XDocument document)
+        {
+            LoadWordsFromDocument(document);
+        }
+
+        protected CodeAnalysisDictionary(IEnumerable<string> recognizedWords)
+        {
+            RecognizedWords.UnionWith(recognizedWords);
```

Comment:
Same here - lets change all the constructors to take the final immutable dictionaries as input parameters.

Reply:
Done